.github/workflows/compound-java-dev.yml

```diff
@@ -0,0 +1,122 @@
+    secrets:
+      docker-password:
+        required: false
+      sonatypeUsername:
+        required: false
+      sonatypePassword:
+        required: false
+      sonatypeSigningKey:
+        required: false
+      sonatypeSigningPassword:
+        required: false
+
+jobs:
+  prebuild-job:
+    name: Prebuild Job
+    uses: th2-net/.github/.github/workflows/compound-prebuild-java-dev-workflow.yml@main
+    with:
+      runsOn: ${{ inputs.runsOn }}
+
+  # if Docker
+  docker-push:
+    name: Build Docker image and push to ghcr.io
+    if: contains(inputs.build-target, 'Docker') || contains(inputs.build-target, 'docker')
+    needs: [prebuild-job]
+    runs-on: ${{ inputs.runsOn }}
+    steps:
+      - name: Show custom release version
+        run: echo ${{ needs.prebuild-job.outputs.version }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ inputs.docker-username }}
+          password: ${{ secrets.docker-password }}
+      - name: Get repository name
+        run: echo "REPOSITORY_NAME=$(echo '${{ github.repository }}' | awk -F '/' '{print $2}')" >> $GITHUB_OUTPUT
+        id: meta
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          tags: ghcr.io/${{ github.repository }}:${{ needs.prebuild-job.outputs.version }}
+          labels: com.exactpro.th2.${{ steps.meta.outputs.REPOSITORY_NAME }}=${{ needs.prebuild-job.outputs.versionNumber }}
+  trivy-docker-scan:
+    name: Scan Docker image for vulnerabilities
+    if: inputs.scanner-enabled
+    needs: [docker-push, prebuild-job]
+    uses: th2-net/.github/.github/workflows/trivy-scan-github.yml@main
+    with:
+      image-path: ghcr.io/${{ github.repository }}:${{ needs.prebuild-job.outputs.version }}
+      target: ${{ inputs.trivy-target }}
+      sensitivity: ${{ inputs.trivy-sensitivity }}
+
+
+# if Sonatype
+  sonatype-push:
+    if: contains(inputs.build-target, 'Sonatype') || contains(inputs.build-target, 'sonatype')
+    name: Build and Push Jar to Sonatype
+    needs: [prebuild-job]
+    runs-on: ${{ inputs.runsOn }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Show custom release version
+        run: echo ${{ needs.prebuild-job.outputs.version }}
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '11'
+      - name: Build with Gradle
+        run: ./gradlew --info clean build publish
+        env:
+          ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.sonatypeUsername }}
+          ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.sonatypePassword }}
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.sonatypeSigningKey }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.sonatypeSigningPassword }}
```

Comment:
Could you please extract to separate first level template (the simplest operation)